js/components/map.js

```diff
@@ -206,8 +202,16 @@ Fliplet.FormBuilder.field('map', {
           }
 
           if (this.suggestionSelected && this.lastChosenAutocompleteValue === value.trim()) {
-            this.value = this.mapField.getTotalAddress();
-            this.updateAddressSuggestions();
+            const timeout = this.mapField.getGeocoder() ? 0 : 3000;
+
+            setTimeout(() => {
+              if (this.mapField.getTotalAddress()) {
+                this.value = this.mapField.getTotalAddress();
+              }
```

Comment:
@armine-fliplet is timeout 0 here for intentional move to the end of event loop? If the timeout is not needed here, i'd go rather with something like:

```suggestion
            const assignAddress = () => {
                 const address = this.mapField.getTotalAddress();
                 if (address) this.value = address;
            }
            
            const geocoderExists = !!this.mapField.getGeocoder();

            if (geocoderExists) assingAddress();
            else setTimeout(assignAddress, 3000);
```

Reply:
@dwfliplet It didn't work without SetTimeout, so I just removed` return`.